src/nvim/mouse.c

```diff
@@ -1259,6 +1259,11 @@ int jump_to_mouse(int flags, bool *inclusive, int which_button)
       col -= curwin->w_grid.col_offset;
     }
 
+    if (col < 0 && curwin->w_leftcol > 0) {
+      curwin->w_leftcol = MAX(0, curwin->w_leftcol + col);
+      leftcol_changed();
+    }
+
```

Comment:
I think the following change can also achieve the same purpose as this, but more consistent as it also scrolls when dragging into the number column:
```diff
diff --git a/src/nvim/mouse.c b/src/nvim/mouse.c
index 95fe4d70d..768611974 100644
--- a/src/nvim/mouse.c
+++ b/src/nvim/mouse.c
@@ -1422,9 +1422,6 @@ bool mouse_comp_pos(win_T *win, int *rowp, int *colp, linenr_T *lnump)
   if (!retval) {
     // Compute the column without wrapping.
     int off = win_col_off(win) - win_col_off2(win);
-    if (col < off) {
-      col = off;
-    }
     col += row * (win->w_width_inner - off);
     // add skip column (for long wrapping line)
     col += win->w_skipcol;
```